### PR TITLE
Coverity issues in framework: algorithms and nexus load*

### PR DESCRIPTION
**Pull request by FedeMPouzols**
Fixes trac issue [#11062](http://trac.mantidproject.org/mantid/ticket/11062).

**To test**

This fixes 14 null dereference issues detected by coverity (see list in ticket description). The issues are in Framework, with a bit of focus on Load***Nexus and related classes.
- Review code
- Check that unit and system tests pass as they seem to do locally.

### CODE REVIEW

File: Code/Mantid/Framework/DataHandling/src/ProcessDasNexusLog.cpp [hunk 1]
```diff
@@ -445,7 +447,9 @@ void ProcessDasNexusLog::writeLogtoFile(API::MatrixWorkspace_sptr ws,
   Kernel::Property *log = ws->run().getProperty(logname);
   Kernel::TimeSeriesProperty<double> *tslog =
       dynamic_cast<Kernel::TimeSeriesProperty<double> *>(log);
-  std::vector<Kernel::DateAndTime> times = tslog->timesAsVector();
+  std::vector<Kernel::DateAndTime> times;
```
Comment (eXeC64) [inline Code/Mantid/Framework/DataHandling/src/ProcessDasNexusLog.cpp]: This won't address the problem fully as tslog is still dereferenced a second time, which is unchecked:
```c++
Kernel::TimeSeriesProperty<double> *tslog = dynamic_cast<Kernel::TimeSeriesProperty<double> *>(log);
std::vector<Kernel::DateAndTime> times;
if (tslog)
  tslog->timesAsVector();
std::vector<double> values = tslog->valuesAsVector();
```

Perhaps instead something like this:
```c++
Kernel::TimeSeriesProperty<double> *tslog = dynamic_cast<Kernel::TimeSeriesProperty<double> *>(log);
if (!tslog)
  throw std::runtime_error("Invalid time series log");
std::vector<Kernel::DateAndTime> times = tslog->timesAsVector();
std::vector<double> values = tslog->valuesAsVector();
```
The exception itself is just a placeholder, but if those values aren't being extracted we're probably wanting to fail.

File: Code/Mantid/Framework/DataHandling/src/LoadNexusLogs.cpp [hunk 2]
```diff
@@ -173,7 +173,9 @@ void LoadNexusLogs::exec() {
       std::vector<Mantid::Kernel::DateAndTime> ptime;
       pval.reserve(event_frame_number.size());
       ptime.reserve(event_frame_number.size());
-      std::vector<Mantid::Kernel::DateAndTime> plogt = plog->timesAsVector();
+      std::vector<Mantid::Kernel::DateAndTime> plogt;
+      if (plog)
+        plogt = plog->timesAsVector();
       std::vector<double> plogv = plog->valuesAsVector();
```
Comment (eXeC64) [inline Code/Mantid/Framework/DataHandling/src/LoadNexusLogs.cpp]: This also has the aforementioned issued - a second dereference that's unchecked.

**Comment by eXeC64** [comment]
Other than the two points I've raised above, the changes look good :)

File: Code/Mantid/Framework/DataHandling/src/ProcessDasNexusLog.cpp (hunk 1 above)
Comment (FedeMPouzols) [inline Code/Mantid/Framework/DataHandling/src/ProcessDasNexusLog.cpp]: Yes, this is better, and raising a exception looks like the proper thing to do. After all tslog is cast from a log entry given by the user, so it should be reported if it's not a valid one.

**Pull request merged**